### ISSUE TITLE
StarshipEngine transposes the sampling results

### ISSUE THREAD

**Issue by josh146** [XanaduAI/strawberryfields#158]
When using the `LocalEngine`, results are returned with shape `[num_shots, num_modes]`. For example, with a 4-mode, 2-shot program:
```python
>>> eng = sf.LocalEngine("gaussian")
>>> results = eng.run(prog)
>>> results.samples
array([[1, 0, 1, 0],
       [0, 1, 0, 1]])
```

However, when using `StarshipEngine`, results are instead returned with shape `[num_modes, num_shots]`. This is the transpose of what is returned from `LocalEngine`. For example, with the same dummy program:

```python
>>> eng = sf.StarshipEngine()
>>> results = eng.run(prog)
>>> results.samples
array([[1, 0],
       [0, 1],
       [1, 0],
       [0, 1]])
```

This is likely a bug, and the samples returned by StarshipEngine need to be transposed before being passed to the user.

**Comment by co9olguy** [comment]
Yep, we should be consistent. Since it is such a simple fix, we can do it right away. 

However, we still need to nail down how we want to pass around and return measurement results in the library and make consistent (see related issue #107, which did not get resolved in time for latest release)

**Comment by zzzeid** [comment]
@josh146 @co9olguy I believe we are wrapping the result *as is* as returned by the API directly in a `Result` object, and there's no intentional transposing happening. Do we want to transpose the result before the API response, or after we get the response?

**Comment by co9olguy** [comment]
ideally the API response should be of the same format as a local backend, so if that's where the discrepency lies, that's where we should fix it (this is still a larger issue through the library tbh)

**Comment by antalszava** [comment]
This was solved by #300
